File: Implementation/Algorithms/cpp/Greedy/Ex03.cpp
/*
 *
 * Problem: https://codeforces.com/problemset/problem/155/A
 *
 */

#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n, c[10000], d, e;
    cin >> n >> c[0];

    d = e = c[0];
    int s = 0;
    for (int i = 1; i < n; ++i) {
        cin >> c[i];

        if (c[i] < d) {
            d = c[i];
            ++s;
        }

        if (c[i] > e) {
            e = c[i];
            ++s;
        }
    }

    cout << s << "\n";

    return 0;
}

File: Implementation/Algorithms/java/Greedy/Ex03.java
/*
 *
 * Problem: https://codeforces.com/problemset/problem/155/A
 *
 */

import java.util.Scanner;

public class Ex03 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n, d, e;
        int[] c = new int[10000];
        
        n = sc.nextInt();
        c[0] = sc.nextInt();

        d = c[0];
        e = c[0];

        int s = 0;
        for (int i = 1; i < n; ++i) {
            c[i] = sc.nextInt();

            if (c[i] > d) {
                d = c[i];
                ++s;
            }

            if (c[i] > e) {
                e = c[i];
                ++s;
            }
        }

        System.out.println(s);

        System.exit(0);
    }
}
